Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -393,7 +394,7 @@ open class BasicFormat: SyntaxRewriter {
       }
     }
 
-    if leadingTrivia.indentation(isOnNewline: previousTokenWillEndWithNewline) == [] {
+    if leadingTrivia.indentation(isOnNewline: isInitialToken || previousTokenWillEndWithNewline) == [] {
```

Comment:
When the formatter takes a node that has a parent node, I think the node should be considered as a starting point for detecting indentation anchor points.

Reply:
Oh, very nice find. I was not expecting the issue in this area. 

This raises a more fundamental question: What does it even mean to format a `CodeBlockSyntax` item standalone while it still has a parent that’s a `FunctionDeclSyntax`? I’m not sure if there really is a correct answer so maybe calling `formatted` should detach the syntax node from the tree, in which case we can always assume that we are formatting an entire tree and not just a subtree. What do you think?

CC @bnbarham: Do you have any opinions on my thoughts above?

Reply:
Thank you for your consideration! I agree with you and have included your test case for now.

Reply:
OK, thank you. Do you want to work on adding 4 spaces in front of `{`? If not, I’ll file an issue for it.

Reply:
Yes! I'd love to work on it 😄

Reply:
That’s great. I’m looking forward to your PR.